backend/Auth/ResetPassword.ts

```diff
@@ -64,3 +68,67 @@ router.post(
     return res.status(204).end();
   }
 );
+
+router.get<Params>(
+  "/validate-token/:token",
+  param("token").exists(),
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return invalidRequest(res, errors);
+    }
+    const token = req.params.token;
+    const tokenType = req.query?.tokenType;
+
+    const rows = await JWTModel.findAll({
+      where: {
+        token: token,
+        jwtType: tokenType,
+      },
+    });
+
+    if (rows.length >= 1) {
+      let token = rows[0].token;
+
+      const userId = rows[0].userId;
+
+      findUserById(userId).then((user) => {
+        const userEmail = user?.email;
+
+        if (tokenIsValid(token, jwtSecret)) {
+          return res.status(200).json({
+            tokenIsValid: true,
+            tokenIsExpired: false,
+            email: userEmail,
+          });
+        } else {
+          return res
+            .status(200)
+            .json({ tokenIsValid: true, tokenIsExpired: true });
+        }
+      });
+    } else {
+      return res
+        .status(200)
+        .json({ tokenIsValid: false, tokenIsExpired: false });
+    }
+  }
+);
+
+function tokenIsValid(token: string, jwtSecret: string | undefined): boolean {
+  if (jwtSecret === undefined) {
```

Comment:
It's a little abnormal to specifically check for undefined rather than just truthiness checks. It implies that `null`, `NaN`, and `""` might be valid JWTs and should proceed to the code below.

Does the `verify()` function throw an error when given `undefined`? Or why was this check needed? If we do need this check, let's use a truthiness check unless `null` or other falsey values are possibly valid jwt tokens:

```suggestion
  if (!jwtSecret) {
```

Reply:
I included the check because it resolved type errors on the Line where verify is called.  The errors were: ```    Argument of type 'string | undefined' is not assignable to parameter of type 'Secret | GetPublicKeyOrSecret'.
      Type 'undefined' is not assignable to type 'Secret | GetPublicKeyOrSecret'.``` Thank you! Adjusted to a truthiness check.